nautobot/extras/tests/test_templatetags.py

```diff
@@ -0,0 +1,197 @@
+"""Test Templatetags provided by Extras."""
+
+from django.contrib.contenttypes.models import ContentType
+from django.test.client import RequestFactory
+from django.urls import reverse
+from nautobot.dcim.models import Site
+from nautobot.extras.models import Job, JobButton
+from nautobot.extras.templatetags.job_buttons import job_buttons
+from nautobot.utilities.testing import TestCase
+
+
+class JobButtonsTest(TestCase):
+    """Test Rendering of Job Buttons."""
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.site = Site.objects.create(name="Test", slug="test")
+        cls.job = Job.objects.get(job_class_name="TestJobButtonReceiverSimple")
+
+        cls.site_type = ContentType.objects.get_for_model(Site)
+
+        job_button_config = {
+            "name": "Job Button Site",
+            "job": cls.job,
+            "defaults": {
+                "text": "Job Button Site",
+                "button_class": "primary",
+            },
+        }
+        cls.jobbutton, _ = JobButton.objects.get_or_create(**job_button_config)
```

Comment:
Since you're modifying and deleting this jobbutton in various tests, this creation may need to be in `setUp()` rather than `setUpTestData()` to avoid leakage of information between tests.

Reply:
Good point since tests don't really run in order.